File: Offer/2209/main.py
from collections import deque


class CQueue:
    stack1: deque[int] 
    stack2: deque[int] 
    def __init__(self):
        self.stack1 = deque([])
        self.stack2 = deque([])

    def appendTail(self, value: int) -> None:
        self.stack1.append(value)

    def deleteHead(self) -> int:
        if len(self.stack2) == 0:
            while len(self.stack1) > 0:
                self.stack2.append(self.stack1.pop())
                
        if len(self.stack2) > 0:
            return self.stack2.pop()
        return -1

File: Offer/2209/main.ts
class CQueue {
    stack1: number[] = []
    stack2: number[] = []
    constructor() {}

    appendTail(value: number): void {
        this.stack1.push(value)
    }

    deleteHead(): number {
        if (this.stack2.length == 0){
            while(this.stack1.length > 0){
                this.stack2.push(this.stack1.pop() ?? 0)
            }
        }
        if (this.stack2.length > 0){
            return this.stack2.pop() ?? 0
        }
        return -1
    }
}